### PR TITLE
accept multiple forms for getMostRecentReport

### LINKED ISSUE #22: Convenient if Utils.getMostRecentReport would accept an array of form names
This is related to #16 .
This will help to avoid having to call this function twice in some situations. An example below illustrates this:

I need to get the latest edd that can either exist in a pregnancy report or an anc report:
This implies that one may write this piece of code:
```
const latestPregnancyReport = Utils.getMostRecentReport('pregnancy')
const latestANCReport = Utils.getMostRecentReport('anc_visit')
if (Utils.isFirstReportNewer(latestANCReport, latestPregnancyReport)) {
  return Utils.getField(latestANCReport, 'edd')
}
return Utils.getField(latestPregnancyReport, 'edd')
```
rather than this:
```
const latestReport = Utils.getMostRecentReport(['pregnancy', 'anc_visit'])
Utils.getField(latestReport, 'edd')
```

### PR DESCRIPTION
**Pull request by freddieptf**
Fixes #22
~~This is a breaking change as i changed the type of one of arguments passed to the function (had some bugs without the type change as both strings and arrays have the `includes` method)~~

### CODE REVIEW

File: src/nootils.js [hunk 1]
```diff
@@ -41,10 +41,11 @@ module.exports = function(settings) {
       return report && report.reported_date;
     },
 
-    getMostRecentReport: function(reports, form, fields) {
+    getMostRecentReport: function(reports, forms, fields) {
+      if (!Array.isArray(forms)) throw 'expecting forms to be a string array';
```
Comment (kennsippell) [inline src/nootils.js:45]: as we discussed today - this needs to be backward compatible

**Review by kennsippell — CHANGES_REQUESTED** [review]

File: src/nootils.js [hunk 2]
```diff
@@ -41,10 +41,11 @@ module.exports = function(settings) {
       return report && report.reported_date;
     },
 
-    getMostRecentReport: function(reports, form, fields) {
+    getMostRecentReport: function(reports, forms, fields) {
+      if (!Array.isArray(forms)) return lib.getMostRecentReport(reports, [forms], fields);
```
Comment (dianabarsan) [inline src/nootils.js:45]: A cleaner way to do this, that doesn't require recursion, and is safer from this point of view, is to do: 
```suggestion
      if (!Array.isArray(forms)) {
        forms = [forms];
      }
```

File: test/nootils.spec.js [hunk 3]
```diff
@@ -99,6 +99,17 @@ describe('Utils', () => {
       expect(actual._id).to.eq(3);
     });
 
+    it('returns most recent matching report from multiple forms', () => {
+      const reports = [
+        { _id: 1, form: 'H', reported_date: 1 },
+        { _id: 2, form: 'V', reported_date: 2 },
+        { _id: 3, form: 'V', reported_date: 3 },
+        { _id: 4, form: 'H', reported_date: 4 }
```
Comment (dianabarsan) [inline test/nootils.spec.js:107]: To improve functional coverage, please add some reports that don't match the forms, so we make sure those are not returned as results. Also, please shuffle `reports` so we don't pass an ordered list. 
Something like: 
```
    const reports = [
        { _id: 6, form: 'B', reported_date: 9 },
        { _id: 3, form: 'V', reported_date: 3 },
        { _id: 4, form: 'H', reported_date: 4 },
        { _id: 2, form: 'V', reported_date: 2 },
        { _id: 5, form: 'A', reported_date: 6 },
        { _id: 1, form: 'H', reported_date: 1 },
        { _id: 7, form: 'C', reported_date: 12 },
      ];
```

**Review by dianabarsan — CHANGES_REQUESTED** [review]
Nice work! I just have a couple of suggestions in line.

**Review by dianabarsan — APPROVED** [review]
Nice work, thanks for taking this on! 

I think we're running into some technical difficulties before this code becomes usable in production: 
1. Travis CI isn't running for the PR, so this can't be merged even if approved. We should migrate to using GHA and update the requirements. 
2. We need to release a new version of cht-nootils and bump the dependency in cht-core for this code to be in effect: https://github.com/medic/cht-core/blob/master/shared-libs/rules-engine/src/rules-emitter.js#L9

**Comment by dianabarsan** [comment]
Can you please try rebasing off of master to get the new CI setup, please?

**Comment by freddieptf** [comment]
Done

**Review by kennsippell — APPROVED** [review]

**Pull request merged**